### PR TITLE
openafs_1_8: include 5.6 fixes

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
cc @lzmartinico

**Comment by Mic92** [comment]
@GrahamcOfBorg eval

File: pkgs/servers/openafs/1.8/module.nix [hunk 1]
```diff
@@ -16,6 +18,22 @@ in stdenv.mkDerivation {
...
+    (fetchpatch {
+      url = "https://github.com/openafs/openafs/commit/77501b48485ed41013a065d05e9b7193884c2267.patch";
+      sha256 = "0rxjqzr8c5ajlk8wrhgjc1qp1934qiriqdi0qxsnk4gj5ibbk4d5";
+    })
+    (fetchpatch {
+      url = "https://github.com/openafs/openafs/commit/355ea43f0d1b7feae1b3af58bc33af12838db7c3.patch";
+      sha256 = "1f9xn8ql6vnxglpj3dvi30sj8vkncazjab2rc13hbw48nvsvcnhm";
+    })
+    (fetchpatch {
+      url = "https://github.com/Mic92/openafs/commit/f7ee87688f6eb4330fb47f3080259e75800312e8.patch";
```
Comment (andersk) [inline pkgs/servers/openafs/1.8/module.nix:32]: Is that different from https://github.com/openafs/openafs/commit/17d38e31e6f2e237a7fb4dfb46841060296310b6.patch?

Comment (Mic92) [reply pkgs/servers/openafs/1.8/module.nix:32]: Ah no. I rebased it, but than figured out did not had to.

File: pkgs/servers/openafs/1.8/module.nix [hunk 2]
```diff
@@ -16,6 +18,22 @@ in stdenv.mkDerivation {
 
   buildInputs = [ kerberos ];
 
+  patches = [
+    # openafs 5.6 patches, included in the next release
+    (fetchpatch {
+      url = "https://github.com/openafs/openafs/commit/77501b48485ed41013a065d05e9b7193884c2267.patch";
```
Comment (andersk) [inline pkgs/servers/openafs/1.8/module.nix:24]: The commit in openafs-stable-1_8_x is https://github.com/openafs/openafs/commit/34f1689b7288688550119638ee9959e453fde414.patch.

Comment (Mic92) [reply pkgs/servers/openafs/1.8/module.nix]: fixed.

File: pkgs/servers/openafs/1.8/module.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/servers/openafs/1.8/module.nix]: fixed.

**Comment by Mic92** [comment]
@GrahamcOfBorg eval

**Comment by Mic92** [comment]
@andersk please test. I already had the backported tested.

**Review by andersk — APPROVED** [review]
Works for me.

**Pull request merged**